File: 1/day1.py
if __name__ == "__main__":
    elf_list = [0]
    for line in open ("inputSimple.txt", "r"):
        if line == "\n":
            elf_list.append (0)
        else:
            elf_list[-1] += int (line)
    elf_list.sort(reverse=True)
    print (elf_list[0])
    print (sum (elf_list[:3]))

File: 1/day1.hs
import qualified Data.List

readLines :: FilePath -> IO [String]
readLines = fmap lines . readFile

convertToInts :: [String] -> [[Int]] -> [[Int]]
convertToInts [] elves =
  elves
convertToInts contents [] =
  convertToInts contents [[]]
convertToInts ("":contents) elves =
  convertToInts contents $ []:elves
convertToInts (line:contents) (elf:elves) =
  convertToInts contents $ ((read line) : elf):elves

sumElves :: [[Int]] -> [Int]
sumElves elves =
  map sum elves

main :: IO ()
main = do
  contents <- readLines ("input.txt")
  let elves = convertToInts contents []
      calElves = sumElves elves
      sortedElves = reverse $ Data.List.sort calElves
    in do
    print $ maximum calElves
    print $ sum $ take 3 sortedElves
